src/sentry/tasks/on_demand_metrics.py

```diff
@@ -252,7 +252,14 @@ def _get_widget_on_demand_specs(
 
     widget_specs = convert_widget_query_to_metric(project_for_query, widget_query, True)
 
-    return widget_specs
+    unique_specs = []
+    hashes = set()
+    for hashed_metric_spec in widget_specs:
```

Comment:
I'm deduplicating here. I could do it instead as part of `convert_widget_query_to_metric`.